### PR TITLE
Fix two build errors: use a static Cast init function and isinstance for type comparison in python 

### PR DESCRIPTION
**Pull request by linzhou-db**
Fix two build errors: 

1. use a static Cast init function to allow delta-sharing-spark work for multiple spark versions. 
2. use isinstance for type comparison in python.

### CODE REVIEW

File: python/delta_sharing/reader.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def _to_pandas(
 
         if for_cdf:
             # Add the change type col name to non cdc actions.
-            if type(action) != AddCdcFile:
```
Comment (zsxwing) [inline python/delta_sharing/reader.py:186]: unrelated change?

**Comment by zsxwing** [comment]
Could you also update the PR title and description?

File: python/delta_sharing/reader.py (hunk 1 above)
Comment (linzhou-db) [reply python/delta_sharing/reader.py:186]: Build and Test /python (3.8) complains about this type comparison so I made the fix.

**Comment by linzhou-db** [comment]
> Could you also update the PR title and description?

updated.

**Review by zsxwing — APPROVED** [review]
LGTM
